sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/KeyVaultKeyStore.java

```diff
@@ -247,13 +255,21 @@ public void engineLoad(InputStream stream, char[] password) {
 
     @Override
     public void engineSetCertificateEntry(String alias, Certificate certificate) {
-        if (aliases == null) {
-            aliases = keyVaultClient.getAliases();
-        }
-        if (!aliases.contains(alias)) {
-            aliases.add(alias);
-            certificates.put(alias, certificate);
+        List<String> portalAlias = keyVaultCertificates.getAliases();
+
+        if (portalAlias != null && portalAlias.contains(alias)) {
+            return;
         }
+        classpathCertificates.setCertificateEntry(alias, certificate);
```

Comment:
It's weird that check by `keyVaultCertificates` but setCertificateEntry by `classpathCertificates`.

Reply:
In my opinion, the user adds a certificate by calling keystore.engineSetCertificateEntry(). If the original certificate (including keyVaultCertificates and classpathCertificates) contains the alias certificate, the insertion fails.
Otherwise, we should not add it to the keyVaultCertificates, because the keyVaultCertificates may be refreshed, resulting in the loss of the certificate added by the user. After being added to the classpathCertificates, when the user obtains the certificate according to the alias, the keyVaultCertificates is first obtained (if the certificate of the alias is added to the keyvault), otherwise, check whether the specified certificate is included in the classpath.

Reply:
Can we check by `classpathCertificates` instead of `keyVaultCertificates`?